corehq/apps/case_search/xpath_functions/utils.py

```diff
@@ -0,0 +1,14 @@
+from django.utils.translation import gettext as _
+
+from eulxml.xpath import serialize
+
+from corehq.apps.case_search.exceptions import XPathFunctionException
+
+
+def confirm_args_count(node, expected):
+    actual = len(node.args)
+    if actual != expected:
+        raise XPathFunctionException(
+            _(f'The "{node.name}" function accepts exactly {expected} arguments, got {actual}'),
+            serialize(node)
+        )
```

Comment:
I initially had aims of putting more such validator functions in here, like confirming that arguments are of the appropriate type (numbers, strings), but didn't want to drop the function-specific context ("the second argument must be a number")

Reply:
I've also wanted to do this. I've been toying with the idea of a decorator that specifies the argument names and types which can be used for validation and type casting.

Reply:
That'd be cool - I thought about doing something like argparse, where you implement a class for each command, but I'm loathe to introduce another inheritance structure.  A decorator would be a nice way to go about it.